### PR TITLE
[REF] Fix more f strings

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to https://github.com/nilearn/nilearn/issues/2528

Changes proposed in this pull request:

- fix remaining f strings
- use flake8-use-fstrings to throw errors in case of usage of `%` or `.format()`

### CODE REVIEW

**Comment by Remi-Gau** [comment]
there are a few string concatenation left here and there in the code base but changing them was actually making it harder to read and they are not flagged as error by flake8-use-fstrings

File: pyproject.toml [hunk 1]
```diff
@@ -93,6 +93,7 @@ dev = [
 	"isort",
 	"flynt",
 	"black",
+  "flake8-use-fstring",
```
Comment (Remi-Gau) [inline pyproject.toml:96]:
```suggestion
   "flake8-use-fstring",
```

**Comment by ymzayek** [comment]
This LGTM! I'm unsure about these azure failures now popping up everywhere. I can try to get an opinion from Azure developers. Let's also see if everything passes on the github tests in https://github.com/nilearn/nilearn/pull/3780

**Comment by ymzayek** [comment]
Something went wrong with the merge here? See diff

**Comment by Remi-Gau** [comment]
> Something went wrong with the merge here? See diff

oops 
indeed

**Comment by Remi-Gau** [comment]
should look better now

File: examples/04_glm_first_level/plot_localizer_surface_analysis.py [hunk 2]
```diff
@@ -211,8 +211,8 @@
 ###############################################################################
 # Finally, we create contrast-specific maps and plot them.
 for index, (contrast_id, contrast_val) in enumerate(contrasts.items()):
-    print('  Contrast % i out of %i: %s, left hemisphere' %
-          (index + 1, len(contrasts), contrast_id))
+    print(f"  Contrast {index + 1:1}% i out of {len(contrasts)}: "
```
Comment (bthirion) [inline examples/04_glm_first_level/plot_localizer_surface_analysis.py:214]:
```suggestion
    print(f"  Contrast {index + 1:1} out of {len(contrasts)}: "
```

Comment (Remi-Gau) [reply examples/04_glm_first_level/plot_localizer_surface_analysis.py:214]: good catch

thanks

**Review by ymzayek — APPROVED** [review]
LGTM thx!

**Pull request merged**